File: hw1pr1.py
import time
import math

start_time = time.time()

### Parse File
fname = "hw1pr1_data.txt"
data = []
with open(fname) as f:
    data = [tuple(map(int, line.split())) for line in f]


### BFS Algorithm
# Distance Formula
def distance(p0, p1):
    return math.sqrt((p0[0] - p1[0])**2 + (p0[1] - p1[1])**2)
# Variables
new_data = []
length = 0
# Start of Algorithm
cPair = data[0]
print (cPair)
new_data.append(cPair)
data.pop(0)
sDistance = 10000000
sIndex = 0
# Run until exhaust data list
while len(data) > 0:
    # search through list for closest
    for i in range(0, len(data)):
        if distance(cPair, data[i]) < sDistance:
            sDistance = distance(cPair, data[i])
            sIndex = i
        else:
            continue
    # Add values to variables and reset for next loop
    cPair = data[sIndex]
    print (cPair)
    new_data.append(cPair)
    data.pop(sIndex)
    length = length + sDistance
    sDistance = 10000000
    sIndex = 0


### Print Total Path Length
print("--- Length : %s ---" % length)


### Print Alloted Time
print("--- %s seconds ---" % (time.time() - start_time))

File: hw1pr1.cpp
#include <fstream>
#include <string>
#include <vector>
#include <math.h>
#include <stdio.h>
#include <iostream>

using namespace std;


struct Node{
  int x;
  int y;

  Node(int x0, int y0)
  {
    x = x0;
    y = y0;
  }
};


int main(){
  ifstream f;
  f.open("hw1pr1_data.txt");
  vector<Node> Nodes;
  int x;
  int y;
  int path = 0;
  double totalDistance = 0;
  int curr = 0;
  int testNode = 0;
  int prev = -1;
  while(f >> x >> y){
      Node n =  Node(x,y);
      Nodes.push_back(n);
  }

  for(int i = 0; i < Nodes.size(); i++){
      path = 0;
      testNode = curr;
      float distance = 1000000000000;
      while(path < Nodes.size()){
        if(path != testNode){
          double testDistance = sqrt(pow((Nodes[testNode].x - Nodes[path].x),2) + pow((Nodes[testNode].y - Nodes[path].y),2));
          if(testDistance < distance && Nodes[path].x != Nodes[prev].x && Nodes[path].y != Nodes[prev].y && (!Nodes[path].visited || path == 0)){
            distance = testDistance;
            curr = path;
          }
        }
        path++;
      }
      totalDistance += distance;
      if(curr == 0){
        break;
      }
      Nodes[curr].visited = true;
      prev = testNode;
  }

  for(int i = 0; i < Nodes.size(); i++){
    cout << Nodes[i].x << " " << Nodes[i].y << endl;
  }

  cout << totalDistance << endl;
}
